### PR TITLE
RawLookup typeahead fix, show only the results from the last 

### PR DESCRIPTION
**Pull request by petrican**
Related to: https://github.com/metasfresh/metasfresh/issues/12169

### CODE REVIEW

File: frontend/src/components/widget/Lookup/RawLookup.js [hunk 1]
```diff
@@ -364,45 +365,50 @@ export class RawLookup extends Component {
...
-        newState.forceEmpty = true;
-        newState.selected = optionNew;
-      } else {
-        list = values;
-        isAlwaysDisplayNewBPartner && list.unshift(optionNew);
+      if (
+        this.typeaheadQuery &&
+        this.typeaheadQuery === typeaheadParams.query
+      ) {
+        let values = response.data.values || [];
```
Comment (teosarca) [inline frontend/src/components/widget/Lookup/RawLookup.js:372]: consider extracting the code starting from this line

File: frontend/src/components/widget/Lookup/RawLookup.js [hunk 2]
```diff
@@ -354,6 +354,7 @@ export class RawLookup extends Component {
         viewId,
       });
     } else {
+      this.typeaheadQuery = typeaheadParams.query;
       typeaheadRequest = autocompleteRequest({
```
Comment (teosarca) [inline frontend/src/components/widget/Lookup/RawLookup.js:358]: make it const here,
or simply inline it.

Comment (teosarca) [reply frontend/src/components/widget/Lookup/RawLookup.js:358]: (same for getViewAttributeTypeahead)

File: frontend/src/components/widget/Lookup/RawLookup.js [hunk 3]
```diff
@@ -354,6 +354,7 @@ export class RawLookup extends Component {
         viewId,
       });
     } else {
+      this.typeaheadQuery = typeaheadParams.query;
```
Comment (teosarca) [inline frontend/src/components/widget/Lookup/RawLookup.js:357]: move it above those "if else if else"

**Comment by petrican** [comment]
@teosarca - I done some more testing after you did the refactoring and this works as expected. Tested also within modals and in other places. Functionality is preserved. I guess this can be merged. let me know. ty

**Review by teosarca — APPROVED** [review]
lgtm

**Pull request merged**